c10/core/DispatchKeySet.h

```diff
@@ -307,10 +607,59 @@ C10_API bool runtimeDispatchKeySetHas(DispatchKey t, DispatchKey k);
 C10_API DispatchKeySet getBackendKeySetFromAutograd(DispatchKey t);
 
 // Returns a DispatchKeySet of autograd related keys mapped to backend.
-C10_API DispatchKeySet getAutogradRelatedKeySetFromBackend(DispatchKey t);
+// for a given backend key, use the associated autograd key.
+// for non-backend keys, use AutogradOther as a default.
+// Note: it's convenient and fast to return a default here rather than (say)
+// returning an optional<DispatchKey>, or throwing. But it makes callers
+// responsible for either a) enforcing the invariant that only backend keys
+// be passed as arguments, or b) interpreting our return value carefully.
+inline DispatchKeySet getAutogradRelatedKeySetFromBackend(BackendBit t, DispatchKeySet ks) {
+  // See Note [Non-Backends that Override Autograd]
+  // If we can get rid of this, then we could drop the ks argument from this function
```

Comment:
That's really sad that we can't do that....

Reply:
update from Alban offline last Friday - we can get rid of this. `AutogradNestedTensor` will be "just another functionality key", and it'll be the job of the nestedtensor extension to register fallthroughs to the `AutogradNestedTensor` key so that nested tensor redispatches to autograd and re-uses existing derivative formulas.